apis/python/src/tiledbsoma/io/_registration/ambient_label_mappings.py

```diff
@@ -661,3 +693,13 @@ def _get_dataframe_joinid_index(df: pd.DataFrame, field_name: str) -> pd.Index:
     if df.index.name in (field_name, "index", None):
         return cast("pd.Index[Any]", df.index)
     raise ValueError(f"Could not find field name {field_name} in dataframe.")
+
+
+def _raise_non_unique_obs_id_error(examples: PDIndex) -> None:
+    a_few_examples = examples[0:4].to_list()
+    msg = f"""Duplicate obs IDs found during registration. {len(examples)} obs IDs are not unique across the provided inputs.
+Example duplicate obs ID(s): {a_few_examples}.
+
+Please ensure obs IDs are unique across all input files for append operations.
```

Comment:
```suggestion
Please ensure obs IDs are unique across all inputs for append operations.
```

Minor tweak since this could be thrown from `register_h5ads` or `register_anndatas`.